adagrams/game.py

```diff
@@ -1,11 +1,273 @@
+import random
+
 def draw_letters():
-    pass
+    
+    letter_pool_str= "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    # List comp for converting ABC string on line 7 into list 
+    letter_pool_list = [letter for letter in letter_pool_str]
+    
+    # Original letter pool dict - stays constant
+    letter_pool_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    # Tracking container: where the letters(key) and their freq (value) will be stored as separate dictionaries
+    tracking_letter_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+
+    # List of randomly drawn hand, stores 10 letters or tiles
+    hand = []
+    
+    # Adding a random letter to the hand, and reducing frequency for each draw
+    while len(hand) < 10:
+        random_letter = letter_pool_list[random.randint(0, 25)]
```

Comment:
Right now your code isn't actually utilizing the weight distribution of the letters. Currently, all letters have the same chance of being selected (they all can only appear their respective limits). `randint` is only randomly selecting from a pool of integers from `0` to `26`. This means that `Z` is just as likely as likely to be selected as `E`. You would need it to select from a pool of integers from `0` to `98` to have the correct distribution.  How could we change your letter pool to consider this weighted distribution?

Reply:
Wow, thank you, you are absolutely right, I had not thought about this.